serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java

```diff
@@ -187,3 +228,106 @@ public E deserializeNullable(@NonNull Decoder decoder, @NonNull DecoderContext c
         return transform(v);
     }
 }
+
+final class EnumValueDeserializer<E extends Enum<E>> implements Deserializer<E> {
+
+    private final Argument<Object> valueType;
+    private final Deserializer<?> valueDeserializer;
+    private final EnumBeanIntrospection<? super E> enumIntrospection;
+    private final Function<E, Object> valueExtractor;
+    private final boolean allowNull;
+
+    public EnumValueDeserializer(Argument<Object> valueType, Deserializer<?> valueDeserializer, EnumBeanIntrospection<? super E> enumIntrospection, Function<E, Object> valueExtractor, boolean allowNull) {
+        this.valueType = valueType;
+        this.valueDeserializer = valueDeserializer;
+        this.enumIntrospection = enumIntrospection;
+        this.valueExtractor = valueExtractor;
+        this.allowNull = allowNull;
+    }
+
+    @NonNull
+    private E transform(@NonNull Decoder decoder, Object value) throws IOException {
+        for (EnumBeanIntrospection.EnumConstant<? super E> enumConstant : enumIntrospection.getConstants()) {
```

Comment:
Can you precalculate some of this stuff? Maybe create a map instead of interesting on every deserialization